src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py

```diff
@@ -821,6 +821,27 @@ def __str__(self):
                 f'\n    spec_location: {self.spec_location}\n')
 
 
+@dataclass
+class SetupParameters:
+    passcode: int
+    vendor_id: int = 0xFFF1
+    product_id: int = 0x8001
+    discriminator: int = 3840
```

Comment:
Should we define those attributes as instance variables instead of class variables?

Reply:
This is a dataclass. The values here are just defaults when creating instance without explicitly setting particular fields:
```python
>>> s1 = SetupParameters(passcode=32)
>>> s2 = SetupParameters(passcode=42, vendor_id=100, product_id=200)
>>> print(s1)
SetupParameters(passcode=32, vendor_id=65521, product_id=32769, ...)
>>> print(s2)
SetupParameters(passcode=42, vendor_id=100, product_id=200, ...)
```

Reply:
Acked, it is marked with @dataclass: